### PR TITLE
LPS-185213

### PR DESCRIPTION
**Pull request by brianchandotcom**

### CODE REVIEW

**Comment by brianchandotcom** [comment]
@pei-jung can you review https://github.com/brianchandotcom/liferay-portal/pull/142274 and resend?

**Comment by locpham97** [comment]
Hi @pei-jung , 
Could you help me review this ticket?
Thanks,
Loc

File: modules/apps/captcha/captcha-rest-impl/src/main/java/com/liferay/captcha/rest/internal/resource/v1_0/CaptchaResourceImpl.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+	private ConfigurationProvider _configurationProvider;
+
+	@Context
+	private Company _contextCompany;
+
+	@Context
+	private HttpServletRequest _httpServletRequest;
+
+	@Context
+	private HttpServletResponse _httpServletResponse;
```
Comment (pei-jung) [inline modules/apps/captcha/captcha-rest-impl/src/main/java/com/liferay/captcha/rest/internal/resource/v1_0/CaptchaResourceImpl.java:130]: These context objects are already set in the base class. You don't need to manually inject them again.

**Review by pei-jung — COMMENTED** [review]
Hi @locpham97, can you clean up the commits a bit and send me another PR? For the captcha configuration, can you drop the commits instead of adding then reverting the changes?  Same thing with the SF commits that removed and added the FFs back. That will help me review the changes faster. Thank you!

File: modules/apps/captcha/captcha-api/src/main/java/com/liferay/captcha/configuration/CaptchaConfiguration.java [hunk 2]
```diff
@@ -17,6 +19,7 @@
 	id = "com.liferay.captcha.configuration.CaptchaConfiguration",
 	localization = "content/Language", name = "captcha-configuration-name"
 )
+@ProviderType
```
[the remark this answers is not in the document]
Comment (alvarosaugarlr) [reply modules/apps/captcha/captcha-api/src/main/java/com/liferay/captcha/configuration/CaptchaConfiguration.java:22]: It was added to indicate how to update the version when using `gw baseline`

**Pull request closed without merging**